src/json.ts

```diff
@@ -123,7 +83,28 @@ export const JsonStruct = define<Json>('Json', (value, context) => {
  * @returns Whether the value is a valid {@link Json} value.
  */
 export function isValidJson(value: unknown): value is Json {
-  return is(value, JsonStruct);
+  try {
+    getValidatedAndSanitizedJson(value);
+    return true;
+  } catch {
+    return false;
+  }
+}
+
+/**
+ * Validate and return sanitized JSON.
+ *
+ * Note:
+ * This function use sanitized JsonStruct for validation
+ * that applies stringify and then parse of a value provided
+ * to ensure that there are no getters which can have side effects
+ * that can cause security issues.
+ *
+ * @param value - JSON structure to be processed.
+ * @returns Sanitized JSON structure.
+ */
+export function getValidatedAndSanitizedJson<Type>(value: unknown): Type {
```

Comment:
```suggestion
export function getValidatedAndSanitizedJson<Type extends Json = Json>(value: unknown): Type {
```

Reply:
I tried adding that but it causes TypeScript errors like:
```
TS2344: Type 'RequestArguments' does not satisfy the constraint 'Json'.   Type 'RequestArguments' is not assignable to type '{ [prop: string]: Json; }'.     Index signature for type 'string' is missing in type 'RequestArguments'.
```
In places like:
```typescript
const sanitizedArgs = getSafeJson<RequestArguments>(args);
```
If used without `RequestArguments`, then problem occurs elsewhere:
```typescript
return await withTeardown(originalRequest(sanitizedArgs), this as any);
```
```
TS2345: Argument of type 'Json' is not assignable to parameter of type 'RequestArguments'.   Type 'null' is not assignable to type 'RequestArguments'.
```
Can be solved with this approach:
```typescript
return await withTeardown(
      originalRequest(sanitizedArgs as unknown as RequestArguments),
     this as any,
);
```
But, it looks worse in my opinion. Is there anything else I can try?

Reply:
I have no strong opinion on this. So, I added as you suggested. If you have a better solution for use cases of this, please leave in the comments on the PR where this is used: https://github.com/MetaMask/snaps-monorepo/pull/1405/files
I will resolve this here, to move this PR forward.